ToyBox/classes/MonkeyPatchin/BagOfPatches/MiscWrath.cs

```diff
@@ -675,6 +677,29 @@ private static void Postfix(Polymorph __instance) {
                 __instance.Owner.View.transform.localScale = new Vector3(scale, scale, scale);
             }
         }
+        [HarmonyPatch(typeof(LocalizedString))]
+        internal class BPTagger 
+        {
+            private static LocalizationPack pack = Kingmaker.Localization.LocalizationManager.LoadPack(Kingmaker.Localization.LocalizationManager.s_CurrentLocale);
+
+            [HarmonyPatch(nameof(LocalizedString.LoadString)), HarmonyPostfix]
+            public static void LoadString_ModTagPatch(ref string __result, LocalizedString __instance) 
+            {
+                if (settings.togglemoddedbptag) 
+                {
+                    try 
+                    {
+                        if (!__result.Contains(".") && !__result.Contains("。")) { return; }
+                        string actualKey = __instance.GetActualKey();
+                        LocalizationPack.StringEntry stringEntry;
+                        if (!pack.m_Strings.TryGetValue(actualKey, out stringEntry) ) 
+                        {
+                            __result += " (modded blueprint)";
+                        }
+                    } catch { }
```

Comment:
When does this throw?
If this *shouldn't* normally throw it might be a good idea to log the exception as a warning.

Reply:
No idea, just stole from String.
Added logging